### PR TITLE
Clippy Update for Release 1.0.9 Branch

### PR DESCRIPTION
**Pull request by yophilav**
- Update Clippy for Release 1.0.9 branch
PS. Thanks to a long bus ride

### CODE REVIEW

**Comment by yophilav** [comment]
I'm running into compile error in VSTS pipeline under `Linux AMD64's Check Submodules`. @arsing pls halp.

**Comment by arsing** [comment]
CI failure is because https://github.com/Azure/iotedge/commit/4f65de285a59cbe73e20cd12e158c787f0ddfea4 needs to be ported too.

File: tools/check_submodules/Cargo.lock [hunk 1]
```diff
@@ -511,81 +491,98 @@ name = "winapi-i686-pc-windows-gnu"
...
+"checksum winapi 0.3.8 (registry+https://github.com/rust-lang/crates.io-index)" = "8093091eeb260906a183e6ae1abdba2ef5ef2257a21801128899c3fc699229c6"
 "checksum winapi-i686-pc-windows-gnu 0.4.0 (registry+https://github.com/rust-lang/crates.io-index)" = "ac3b87c63620426dd9b991e5ce0329eff545bccbbb34f3be09ff6fb6ab51b7b6"
+"checksum winapi-util 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)" = "7168bab6e1daee33b4557efd0e95d5ca70a03706d39fa5f3fe7a236f584b03c9"
 "checksum winapi-x86_64-pc-windows-gnu 0.4.0 (registry+https://github.com/rust-lang/crates.io-index)" = "712e227841d057c1ee1cd2fb22fa7e5a5461ae8e48fa2ca79ec42cfc1931183f"
-"checksum wincolor 0.1.6 (registry+https://github.com/rust-lang/crates.io-index)" = "eeb06499a3a4d44302791052df005d5232b927ed1a9658146d842165c4de7767"
+"checksum wincolor 1.0.2 (registry+https://github.com/rust-lang/crates.io-index)" = "96f5016b18804d24db43cebf3c77269e7569b8954a8464501c216cc5e070eaa9"
+"checksum yaml-rust 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)" = "65923dd1784f44da1d2c3dbbc5e822045628c590ba72123e1c73d3c230c4434d"
```
Comment (arsing) [inline tools/check_submodules/Cargo.lock]: File is not identical to master. You're missing a trailing newline.

File: tools/check_submodules/Cargo.toml [hunk 2]
```diff
@@ -12,4 +12,4 @@ git2 = "0.7"
 hex = "0.3"
 log = "0.4"
 
-edgelet-utils = { path = "../../edgelet/edgelet-utils" }
+edgelet-utils = { path = "../../edgelet/edgelet-utils" }
```
Comment (arsing) [inline tools/check_submodules/Cargo.toml]: Same here, the file in master has a trailing newline.

File: tools/check_submodules/Cargo.lock (hunk 1 above)
Comment (yophilav) [reply tools/check_submodules/Cargo.lock]: The missing newline characters have returned.

File: tools/check_submodules/Cargo.toml (hunk 2 above)
Comment (yophilav) [reply tools/check_submodules/Cargo.toml]: The missing newline characters have returned.

**Pull request merged**